### PR TITLE
chore(curriculum): update tests lab product page

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

I've cleaned up the assertion methods in the lab product landing page.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 1]
```diff
@@ -35,67 +35,76 @@ Fulfill the user stories below and get all the tests to pass to complete the lab
...
+assert.strictEqual(el.tagName,'IMG'); 
 ```
 
 Your `#header-img` should be a descendant of the `#header`.
 
 ```js
-const els = document.querySelectorAll('#header #header-img')
-assert(els.length > 0)
+const els = document.querySelectorAll('#header #header-img');
+assert.isAbove(els.length,0);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:55]:
```suggestion
assert.isNotEmpty(els);
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 2]
```diff
@@ -104,16 +113,20 @@ Each `.nav-link` element should link to a corresponding element on the landing p
 const els = document.querySelectorAll('.nav-link')
 els.forEach(el => {
   const linkDestination = el.getAttribute('href').slice(1)
-  if (!document.getElementById(linkDestination)) assert(false)
+  if (!document.getElementById(linkDestination)) 
+  {
+    assert.fail(); 
+  }
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:119]:
```suggestion
  assert.exists(document.getElementById(linkDestination));
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 3]
```diff
@@ -104,16 +113,20 @@ Each `.nav-link` element should link to a corresponding element on the landing p
...
 })
-assert(els.length > 0)
+assert.isAbove(els.length,0);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:121]:
```suggestion
assert.isNotEmpty(els); 
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 4]
```diff
@@ -104,16 +113,20 @@ Each `.nav-link` element should link to a corresponding element on the landing p
...
 ```
 
 You should have a `video` or `iframe` element with an `id` of `video`.
 
 ```js
-const el = document.getElementById('video')
-assert(!!el && (el.tagName === 'VIDEO' || el.tagName === 'IFRAME'))
+const el = document.getElementById('video');
+assert.exists(el); 
+assert.oneOf(el.tagNam, ['VIDEO','IFRAME']);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:129]:
```suggestion
assert.oneOf(el.tagName, ['VIDEO','IFRAME']);
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 5]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 You should have a `form` element with an `id` of `form`.
 
 ```js
-const el = document.getElementById('form')
-assert(!!el && el.tagName === 'FORM')
+const el = document.getElementById('form');
+assert.exists(el); 
+assert.strictEqual(el.tagName,"FORM");
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:152]:
```suggestion
assert.strictEqual(el.tagName, "FORM"); 
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 6]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 You should have an `input` element with an `id` of `email`.
 
 ```js
-const el = document.getElementById('email')
-assert(!!el && el.tagName === 'INPUT')
+const el = document.getElementById('email');
+assert.exists(el); 
+assert.strictEqual(el.tagName,"INPUT");
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:160]:
```suggestion
assert.strictEqual(el.tagName, "INPUT"); 
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 7]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
+assert.exists(el); 
+assert.strictEqual(el.tagName,"INPUT"); 
 ```
 
 Your `#email` should be a descendant of the `#form`.
 
 ```js
 const els = document.querySelectorAll('#form #email')
-assert(els.length > 0)
+assert.isAbove(els.length,0);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:167]:
```suggestion
assert.isNotEmpty(els);
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 8]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 Your `#email` should use HTML5 validation by setting its `type` to `email`.
 
 ```js
-const el = document.getElementById('email')
-assert(!!el && el.type === 'email')
+const el = document.getElementById('email');
+assert.exists(el); 
+assert.strictEqual(el.type,"email");
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:184]:
```suggestion
assert.strictEqual(el.type, "email");
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 9]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 You should have an `input` element with an `id` of `submit`.
 
 ```js
-const el = document.getElementById('submit')
-assert(!!el && el.tagName === 'INPUT')
+const el = document.getElementById('submit');
+assert.exists(el); 
+assert.strictEqual(el.tagName,"INPUT");
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:192]:
```suggestion
assert.strictEqual(el.tagName, "INPUT");
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 10]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 Your `#submit` should be a descendant of the `#form`.
 
 ```js
-const els = document.querySelectorAll('#form #submit')
-assert(els.length > 0)
+const els = document.querySelectorAll('#form #submit');
+assert.isAbove(els.length,0);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:199]:
```suggestion
assert.isNotEmpty(els);
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 11]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 Your `#submit` should have a `type` of `submit`.
 
 ```js
-const el = document.getElementById('submit')
-assert(!!el && el.type === 'submit')
+const el = document.getElementById('submit');
+assert.exists(el); 
+assert.strictEqual(el.type,'submit');
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:207]:
```suggestion
assert.strictEqual(el.type, 'submit');
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 12]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 Your `#form` should have an `action` attribute of `https://www.freecodecamp.org/email-submit`.
 
 ```js
-const el = document.getElementById('form')
-assert(!!el && el.action === 'https://www.freecodecamp.org/email-submit')
+const el = document.getElementById('form');
+assert.exists(el); 
+assert.strictEqual(el.action,'https://www.freecodecamp.org/email-submit');
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:215]:
```suggestion
assert.strictEqual(el.action, 'https://www.freecodecamp.org/email-submit');
```

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 13]
```diff
@@ -128,77 +141,86 @@ if (sourceNode.length) {
...
 ```
 
 Your `#email` should have a `name` attribute of `email`.
 
 ```js
-const el = document.getElementById('email')
-assert(!!el && el.name === 'email')
+const el = document.getElementById('email');
+assert.exists(el); 
+assert.strictEqual(el.name,'email');
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:223]:
```suggestion
assert.strictEqual(el.name, 'email');
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 14]
```diff
@@ -35,67 +35,76 @@ Fulfill the user stories below and get all the tests to pass to complete the lab
...
+assert.exists(el.src); 
 ```
 
 Your `#header-img`’s `src` value should be a valid URL (starts with `http`).
 
 ```js
 const el = document.getElementById('header-img');
-assert(!!el && el.getAttribute('src') !== "" && /^http/.test(el.src));
+assert.exists(el);
+assert.notEmpty(el.getAttribute('src'));
```
[the remark this answers is not in the document]
Comment (a2937) [reply curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:71]: I don't understand why the tests didn't fail. I mean yes, I did something wrong. But I have a tendency to count on tests to tell me when I make typos like that. 

image

**Review by majestic-owl448 — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md [hunk 15]
```diff
@@ -260,7 +277,7 @@ const usesFlex = cssRules.find(rule => hasFlex(rule))
 const usesFlexMedia = mediaRules.find(mediaRule => {
   return [...mediaRule.cssRules].find(rule => hasFlex(rule))
 })
-assert(usesFlex || usesFlexMedia)
+assert.isTrue(usesFlex || usesFlexMedia);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-product-landing-page/587d78af367417b2b2512b04.md:280]:
```suggestion
assert.exists(usesFlex || usesFlexMedia);
```

With `isTrue` it's failing as `find` returns the found element or `undefined`, so here something like `isDefined` or `exists` would work better than `isTrue`

**Review by moT01 — APPROVED** [review]

**Pull request merged**